market-generator/src/main/java/com/market/banica/generator/configuration/MarketConfigurationImpl.java

```diff
@@ -0,0 +1,167 @@
+import com.market.banica.generator.property.LinkedProperties;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.EnableMBeanExport;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedResource;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.BiConsumer;
+
+@Component
+@ManagedResource
+@EnableMBeanExport
+@ConfigurationProperties
+public class MarketConfigurationImpl implements MarketConfiguration {
+    private static final String ORIGIN_GOOD_PATTERN = "%s.%s";
+
+    private final File file;
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+
+    private final Map<String, GoodSpecification> goods = new HashMap<>();
+
+//    TODO: ADD TickGenerator when it's ready
+
+    @Autowired
+    public MarketConfigurationImpl(@Value("${market.properties.file.path}") String path) {
+        this.file = new File(path);
+    }
+
+    @Override
+    @ManagedOperation
+    public void addGoodSpecification(String origin, String good,
+                                     long quantityLow, long quantityHigh, long quantityStep,
+                                     double priceLow, double priceHigh, double priceStep,
+                                     int periodLow, int periodHigh, int periodStep) {
+
+        try {
+            this.lock.writeLock().lock();
+
+            origin = origin.toLowerCase();
+            good = good.toLowerCase();
+
+            if (doesGoodExist(String.format(ORIGIN_GOOD_PATTERN, origin, good))) {
+                throw new IllegalArgumentException(String.format("A good with name %s from %s already exists",
+                        good.toUpperCase(), origin.toUpperCase()));
+            }
+
+            Properties properties = new LinkedProperties();
+
+            GoodSpecification goodSpec = new GoodSpecification(good, quantityLow, quantityHigh, quantityStep,
+                    priceLow, priceHigh, priceStep, periodLow, periodHigh, periodStep);
+
+            this.modifyProperties(goodSpec.generateProperties(origin), properties,
+                    properties::setProperty, this.file, false);
+            this.goods.put(String.format(ORIGIN_GOOD_PATTERN, origin, good), goodSpec);
+        } finally {
+            this.lock.writeLock().unlock();
+        }
+
+
+//        System.out.println("Adding"); TODO: replace writing in console with Logging
```

Comment:
is this going to be done in a separate task? if it isn't - could you please add the logging or at least leave only the TODO part without this commented system out